### PR TITLE
Update to Symfony 4.4

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Toflar) [inline .github/workflows/ci.yml]: Can you explain this?

Comment (leofeyer) [reply .github/workflows/ci.yml]: I knew you would stumble upon it. 😄 

The functional tests currently fail, because of a third-party deprecation that is incorrectly treated as a "self" deprecation:

```
Remaining self deprecation notices (124)

  86x: The "twig.exception_listener" service is deprecated since Symfony 4.4.
    23x in RoutingTest::testResolvesAliases from Contao\CoreBundle\Tests\Functional
    22x in RoutingTest::testResolvesAliasesWithLocale from Contao\CoreBundle\Tests\Functional
    19x in RoutingTest::testResolvesAliasesWithoutUrlSuffix from Contao\CoreBundle\Tests\Functional
    13x in RoutingTest::testResolvesTheRootPageWithLocale from Contao\CoreBundle\Tests\Functional
    8x in RoutingTest::testResolvesTheRootPage from Contao\CoreBundle\Tests\Functional
    1x in RoutingTest::testOrdersThePageModelsByCandidates from Contao\CoreBundle\Tests\Functional

  38x: The "twig.controller.exception" service is deprecated since Symfony 4.4.
    13x in RoutingTest::testResolvesAliases from Contao\CoreBundle\Tests\Functional
    12x in RoutingTest::testResolvesAliasesWithLocale from Contao\CoreBundle\Tests\Functional
    9x in RoutingTest::testResolvesAliasesWithoutUrlSuffix from Contao\CoreBundle\Tests\Functional
    3x in RoutingTest::testResolvesTheRootPageWithLocale from Contao\CoreBundle\Tests\Functional
    1x in RoutingTest::testResolvesTheRootPage from Contao\CoreBundle\Tests\Functional
```

Both the `twig.exception_listener` and the `twig.controller.exception` services use deprecated classes. However, since these services are defined by Symfony and not by Contao, there is no way for us to change them. (I have also tried to overwrite the service definitions, however, then the functional tests no longer work, because other parts of Symfony still expect the old, deprecated class.)

So until this issue has been fixed in Symfony, disabling the deprecation helper for the functional tests seemed the best option. Note that the deprecation helper is still enabled for the unit tests, so most of the deprecations are still reported.

Comment (Toflar) [reply .github/workflows/ci.yml]: Imho it should be configured to `max[self]=0`. Doesn‘t that work?

Comment (leofeyer) [reply .github/workflows/ci.yml]: No, because that translates to "fail if there are more than 0 self deprecations". It would work with `max[self]=125`, because we currently have 124 self deprecations. But there is no point in that.

Comment (Toflar) [reply .github/workflows/ci.yml]: Why do we have self deprecations that are not fixed?😱

Comment (leofeyer) [reply .github/workflows/ci.yml]: I have explained this in detail above.

Comment (Toflar) [reply .github/workflows/ci.yml]: Okay, I have to check that in more detail but there‘s no way we can merge that like that.

**Review by Toflar — CHANGES_REQUESTED** [review]
Cannot merge until the deprecations situation is clarified.

File: .github/workflows/ci.yml
Comment (leofeyer) [reply .github/workflows/ci.yml]: Why not? We are already notified about deprecations in the unit tests. No need to run them all again in the functional tests, is there?

Comment (leofeyer) [reply .github/workflows/ci.yml]: We can add a `# FIXME` if it makes you feel better. But we really need to merge this PR, because I cannot continue working on Contao 4.9 without it.

Comment (Toflar) [reply .github/workflows/ci.yml]: Yeah I don't want to have this merged because we might forget about it and then stop fixing our deprecations :) 
But again: You state the following

> The functional tests currently fail, because of a third-party deprecation that is incorrectly treated as a "self" deprecation:

Is this a bug that's been worked on or why does that happen? Maybe it's something we can fix?

Comment (leofeyer) [reply .github/workflows/ci.yml]: We cannot (in Contao). As I have explained above

> Both the `twig.exception_listener` and the `twig.controller.exception` services use deprecated classes.

The bug is in the Symfony PHPUnit bridge, which considers the deprecated classes "self deprecations", although they are third-party deprecations.

**Comment by leofeyer** [comment]
Again, it is vital that we merge this PR as soon as possible, because **no other PR can be merged** without the fixes from this one (see e.g. https://github.com/contao/contao/pull/621#issuecomment-557835027).

**Comment by m-vo** [comment]
@leofeyer Did you already create an issue at Symfony / PHPUnit bridge? 

I don't see a problem merging this PR as long as we create a marker issue (bug) to reenable the deprecation helper again.

File: .github/workflows/ci.yml
Comment (Toflar) [reply .github/workflows/ci.yml]: Yes, I did understand this. But then we have to report it and add the issue in the FIXME description and not wait for it to randomly vanish?

**Comment by leofeyer** [comment]
I have fixed the deprecations problem in bd93c66c693d2c73afb683a664939fa86b2c8b57. Turns out it was neither a bug in the PHPUnit bridge nor in Contao.

The PHPUnit bridge decides based on the file path whether an exception is a self, a direct or an indirect exception. This works well with the default Symfony directory structure:

```
bundle/
    src/
    tests/
    var/cache/test/
```

However, we had the functional tests in a subfolder of the `tests/` folder:

```
core-bundle/
    src/
    tests/
        Functional/
            var/cache/test/
```

So when comparing paths, the PHPUnit bridge – more or less correctly – decided that the deprecations are thrown in our own code and are therefore self deprecations.

**Review by Toflar — APPROVED** [review]

**Review by bytehead — APPROVED** [review]

**Pull request merged**
